core/src/main/java/org/apache/druid/java/util/common/HumanReadableBytes.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.common;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@JsonSerialize(using = HumanReadableBytesSerializer.class)
+public class HumanReadableBytes
+{
+  public static final HumanReadableBytes ZERO = new HumanReadableBytes(0L);
+
+  private final long bytes;
+
+  public HumanReadableBytes(String bytes)
+  {
+    this.bytes = HumanReadableBytes.parse(bytes);
+  }
+
+  public HumanReadableBytes(long bytes)
+  {
+    this.bytes = bytes;
+  }
+
+  public long getBytes()
+  {
+    return bytes;
+  }
+
+  public int getBytesInInt()
+  {
+    if (bytes > Integer.MAX_VALUE) {
+      throw new ISE("Number overflow");
+    }
+
+    return (int) bytes;
+  }
+
+  @Override
+  public boolean equals(Object thatObj)
+  {
+    if (thatObj == null) {
+      return false;
+    }
+    if (thatObj instanceof HumanReadableBytes) {
+      return bytes == ((HumanReadableBytes) thatObj).bytes;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Long.hashCode(bytes);
+  }
+
+  @Override
+  public String toString()
+  {
+    return String.valueOf(bytes);
+  }
+
+  public static HumanReadableBytes valueOf(int bytes)
+  {
+    return new HumanReadableBytes(bytes);
+  }
+
+  public static HumanReadableBytes valueOf(long bytes)
+  {
+    return new HumanReadableBytes(bytes);
+  }
+
+  public static long parse(String number)
+  {
+    if (number == null) {
+      throw new IAE("Invalid format of number: number is null");
+    }
+
+    number = number.trim();
+    if (number.length() == 0) {
+      throw new IAE("Invalid format of number: Number is empty");
```

Comment:
The error message is best to be consistent with the previous one. In addition, perhaps it is more appropriate to use `blank` instead of `empty` here.